src/components/ContentOpenWith/IntegrationPortal.js

```diff
@@ -4,69 +4,51 @@
  * @author Box
  */
 
-import { PureComponent } from 'react';
 import { createPortal } from 'react-dom';
 
 type Props = {
     children: any,
     integrationWindow: any,
 };
-class IntegrationPortal extends PureComponent<Props> {
-    containerElement: ?HTMLElement;
 
-    /**
-     * [constructor]
-     *
-     * @private
-     * @return {IntegrationPortal}
-     */
-    constructor(props: Props) {
-        super(props);
-
-        const { integrationWindow }: Props = this.props;
-        this.containerElement = integrationWindow.document.createElement('div');
-    }
-
-    /**
-     * Setup UI in the new window.
-     *
-     * @private
-     * @inheritdoc
-     * @return {void}
-     */
-    componentDidMount() {
```

Comment:
It's usually a convention to do DOM manipulation within `componentDidMount`. What's the reasoning for moving to a function?

Reply:
If you portal a component to a new tab and close that tab, it does not seem to unmount from React’s perspective.

Reply:
Hm, okay. I think you'd still want to follow the convention, though. If you need to force the component to remount, can you use a `key` prop to ensure it's unique?

Reply:
fair enough, I'll give that a shot.